cloud_storage_handler/main.py

```diff
@@ -39,13 +29,50 @@ def init_app() -> FlaskApp:
 
     foca = Foca(
         config_file=config_path,
+        custom_config_model="cloud_storage_handler.custom_config.CustomConfig",
     )
-    return foca.create_app()
+    foca_config = foca.conf
+    minio_config = foca_config.custom.minio
+    logger.info(minio_config)
+
+    # Create the Connexion FlaskApp instance
+    return foca.create_app(), minio_config
 
 
 def main() -> None:
     """Run FOCA application."""
-    app = init_app()
+    app, minio_config = init_app()
+    foca_app = app.app
+
+    if minio_config is None:
+        raise KeyError("Custom configuration for MinIO not found in Flask app config.")
+
+    # Initialize MinIO client
+    try:
+        minio_client = Minio(
+            endpoint=f"{minio_config.hostname}:{minio_config.port}",
+            access_key=minio_config.access_key,
+            secret_key=minio_config.secret_key,
+            secure=minio_config.secure,
+        )
+        foca_app.config["minio_client"] = minio_client
```

Comment:
I will need the minio client in the controller to access some functions for some apis like list files.

Reply:
I agree with this general idea, but I think it would be better to formalize this by defining an optional `instance` parameter in `MinioConfig` with default `None`.